packages/office-ui-fabric-react/src/components/MarqueeSelection/MarqueeSelection.base.tsx

```diff
@@ -216,22 +217,52 @@ export class MarqueeSelectionBase extends BaseComponent<IMarqueeSelectionProps,
 
           this._preservedIndicies = selection && selection.getSelectedIndices && selection.getSelectedIndices();
         }
+
         // We need to constrain the current point to the rootRect boundaries.
-        const constrainedPoint = this.props.isDraggingConstrainedToRoot
-          ? {
-              x: Math.max(0, Math.min(rootRect.width, this._lastMouseEvent!.clientX - rootRect.left)),
-              y: Math.max(0, Math.min(rootRect.height, this._lastMouseEvent!.clientY - rootRect.top))
-            }
-          : {
-              x: this._lastMouseEvent!.clientX - rootRect.left,
-              y: this._lastMouseEvent!.clientY - rootRect.top
-            };
+        // and constrain dragRect within viewport
+        const constrainedPoint = {
+          x: this._lastMouseEvent!.clientX - rootRect.left,
+          y: this._lastMouseEvent!.clientY - rootRect.top
+        };
+
+        let rectLeft = Math.min(this._dragOrigin.x, constrainedPoint.x);
+        let rectTop = Math.min(this._dragOrigin.y, constrainedPoint.y);
+
+        let rectWidth = Math.abs(constrainedPoint.x - this._dragOrigin.x);
+        let rectHeight = Math.abs(constrainedPoint.y - this._dragOrigin.y);
+
+        if (rectLeft < 0) {
+          rectLeft = 0;
+          rectWidth = this._dragOrigin.x;
+        }
+
+        if (rectTop < 0) {
+          rectTop = 0;
+          rectHeight = this._dragOrigin.y;
+        }
+
+        // constrain rect height to root rect height, if root rect height is valid.
+        if (rootRect.width > 0) {
+          if (rectTop + rectHeight > rootRect.height) {
+            rectHeight = rootRect.height - rectTop;
+          }
+        }
+
+        // constrain rect width to root rect width, if root rect width is valid.
+        if (rootRect.width > 0) {
+          // since we don't have a width pass down to the element, we use the _scrollableSurface width
+          // to constraint the marquee rect to not falls out the right boundary
+          const viewportWidth = rootRect.width > 0 ? rootRect.width : this._scrollableSurface!.clientWidth;
+          if (rectLeft + rectWidth > viewportWidth) {
+            rectWidth = viewportWidth - rectLeft;
+          }
+        }
 
         const dragRect = {
```

Comment:
i'm not totally sure I follow the before/after behavioral change here. Maybe you could add a test to illustrate what behavior didn't work before that now works.

Reply:
I have removed this change.